packages/cardano-services/src/util/polling/DbSyncEpochPollService.ts

```diff
@@ -0,0 +1,62 @@
+import { EpochModel } from '../../StakePool';
+import { InMemoryCache, UNLIMITED_CACHE_TTL } from '../../InMemoryCache';
+import { Pool, QueryResult } from 'pg';
+import { findLastEpoch } from './queries';
+
+export const EPOCH_POLL_INTERVAL_DEFAULT = 10_000;
+export const CURRENT_EPOCH_CACHE_KEY = 'current_epoch';
+
+/**
+ * Class to handle epoch rollover through db polling
+ */
+export class DbSyncEpochPollService {
```

Comment:
Would be nice to also export an interface that dependents can depend on instead of this specific class. Maybe move cache clearing logic out of it too? Otherwise it should be callled EpochRolloverCacheCleaner or sth. How about
```suggestion
export interface EpochMonitor {
  onEpochRollover(callback: (epoch: Cardano.Epoch) => void));
}
export class DbSyncEpochPollService implements EpochMonitor {
```

Or just `Observable<Cardano.Epoch>`, but cardano-services does not depend on rxjs so maybe not the best idea to add this dependency just for this :sweat_smile: 

Thoughts @rhyslbw @IvayloAndonov ?

Reply:
I like the interface idea as a way to complete this initial feature. I think we should be looking at Observerable patterns in `cardano-services` too, and this is a good candidate, but it would be best to assess the whole design and plan it out as a refactor when we have capacity.

Reply:
`EpochMonitor` interface as dependency is a good suggestion, but I'm not so competent on the idea of Observable usage with `rxjs` though

Reply:
Let's keep rxjs out of it for now as @rhyslbw suggested, let's just apply the pattern with callbacks

Reply:
Callbacks patter is applied under https://github.com/input-output-hk/cardano-js-sdk/pull/370/commits/9a6e57e6440feb8df6d8ccaaeaa681c64f5da672 @mkazlauskas